src/contracts/middleware/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -0,0 +1,555 @@
+    }
+
+    /// @return numOperatorsPerQuorum is the list of number of operators per quorum in quorumNumberss
+    function _registerOperatorWithCoordinator(address operator, bytes calldata quorumNumbers, BN254.G1Point memory pubkey, string memory socket) internal returns(uint32[] memory) {
+        // require(
+        //     slasher.contractCanSlashOperatorUntilBlock(operator, address(serviceManager)) == type(uint32).max,
+        //     "StakeRegistry._registerOperator: operator must be opted into slashing by the serviceManager"
+        // );
+        
+        // get the quorum bitmap from the quorum numbers
+        uint256 quorumBitmap = BitmapUtils.orderedBytesArrayToBitmap(quorumNumbers);
+        require(quorumBitmap <= MiddlewareUtils.MAX_QUORUM_BITMAP, "BLSRegistryCoordinatorWithIndices._registerOperatorWithCoordinator: quorumBitmap exceeds of max bitmap size");
+        require(quorumBitmap != 0, "BLSRegistryCoordinatorWithIndices._registerOperatorWithCoordinator: quorumBitmap cannot be 0");
+        // register the operator with the BLSPubkeyRegistry and get the operatorId (in this case, the pubkeyHash) back
+        bytes32 operatorId = blsPubkeyRegistry.registerOperator(operator, quorumNumbers, pubkey);
+
+        uint256 operatorQuorumBitmapHistoryLength = _operatorIdToQuorumBitmapHistory[operatorId].length;
+        if(operatorQuorumBitmapHistoryLength > 0) {
+            uint256 prevQuorumBitmap = _operatorIdToQuorumBitmapHistory[operatorId][operatorQuorumBitmapHistoryLength - 1].quorumBitmap;
+            require(prevQuorumBitmap & quorumBitmap == 0, "BLSRegistryCoordinatorWithIndices._registerOperatorWithCoordinator: operator already registered for some quorums being registered for");
+            // new stored quorumBitmap is the previous quorumBitmap or'd with the new quorumBitmap to register for
+            quorumBitmap |= prevQuorumBitmap;
+        }
+
+        // register the operator with the StakeRegistry
+        stakeRegistry.registerOperator(operator, operatorId, quorumNumbers);
+
+        // register the operator with the IndexRegistry
+        uint32[] memory numOperatorsPerQuorum = indexRegistry.registerOperator(operatorId, quorumNumbers);
+
+        // set the operatorId to quorum bitmap history
+        _operatorIdToQuorumBitmapHistory[operatorId].push(QuorumBitmapUpdate({
+            updateBlockNumber: uint32(block.number),
+            nextUpdateBlockNumber: 0,
+            quorumBitmap: uint192(quorumBitmap)
+        }));
+
+        // set the operator struct
+        _operators[operator] = Operator({
+            operatorId: operatorId,
+            status: OperatorStatus.REGISTERED
+        });
+
+        // record a stake update not bonding the operator at all (unbonded at 0), because they haven't served anything yet
+        // serviceManager.recordFirstStakeUpdate(operator, 0);
+
+        emit OperatorRegistered(operator, operatorId);
+
+        emit OperatorSocketUpdate(operatorId, socket);
```

Comment:
can this function (`_registerOperatorWithCoordinator`) be called when the operator is already registered for some quorums?
seems to me like it can be (and I think it's supposed to be)?
documentation should make intended behavior here clear -- TODO for myself @ChaoticWalrus  to check the docs PR for this.
I'm somewhat concerned about the highlighted code being run when the operator is already registered. here's a few questions:

- we aren't updating the previous entry in the "quorum bitmap history" -- is this wrong? seems like we could end up in a case where multiple entries have `nextUpdateBlockNumber = 0`, which would be a problem AFAIK
- we're setting the operator struct here -- presumably the state being set to REGISTERED is fine, am pretty sure the operatorId can't change here, just want to be certain (seems like potentially an unstated but used property of this system)
- do we want the events here? I guess the socket update is happening, so it makes sense, and the `OperatorRegistered` event could trigger logic to check what (new) quorums the operator is registered for?

Reply:
@gpsanant see the above comment -- I think the concern about ending up with multiple entries in the bitmap history that have `nextUpdateBlockNumber = 0` is the most serious here.

Reply:
great point. updates here: https://github.com/Layr-Labs/eigenlayer-contracts/pull/35/commits/cf6e343b6542825ece78995d7a3e47284f5ef3b5

Reply:
LGTM, ty ty. And you're happy with the events emitted here?